astr303a.py

```diff
@@ -0,0 +1,571 @@
+    plt.annotate(r'$\bar z$'+' ='+str('{:.4E}'.format(coeffs[1]))+'\n'+r'$\sigma_z$'+' ='+str('{:.4E}'.format(abs(coeffs[2]))), xy=(0.70, 0.90), xycoords='axes fraction',fontsize = 10)
+    pp.savefig()
+    plt.clf()
+
+    print('Gaussian fit: peak,mean,sigma',coeffs)
+    return coeffs #return gaussian fitting
+
+#Compute recessional velocity and distance from Hubble's law.
+def part5(gausscoeffs,dataarray):
+
+    #Compute recessional velocity 
+    zcoma = gausscoeffs[1]
+    sigmacoma = gausscoeffs[2]
+    c = 2.99792485e8
+    vcoma = c*(((zcoma+1)**2)-1)/(((zcoma+1)**2)+1)
+    vcoma = vcoma/1000
+
+    #Compute distance from Hubble law
+    H = 67.8
+    dcoma = vcoma/H
+    
+    print('Coma radial velocity: '+str(vcoma)+'km/s')
+    print('Coma distance (from Hubble law): '+str(dcoma)+'Mpc')
+    return dcoma,zcoma #return cluster distance and mean redshift
+
+#Define function for computing angular distance between two points
+def projecteddistance(srcRA,srcDEC,comaRA,comaDEC,dcoma):
+
+    #convert degrees to radians
+    srcRArad = np.pi*srcRA/180
+    srcDECrad = np.pi*srcDEC/180
+    comaRArad = np.pi*comaRA/180
+    comaDECrad = np.pi*comaDEC/180
+
+    #Computation of angular distance  
+    term1 = (np.sin(srcDECrad)*np.sin(comaDECrad))
+    term2 = (np.cos(srcDECrad)*np.cos(comaDECrad)*np.cos((srcRArad)-(comaRArad)))
+    combin = round(term1+term2,15) #round to 15 places
+    radians = abs(np.arccos(combin))
+    return radians
+
+#Coma Cluster member selection algorithm
+def part678(comaRA,comaDEC,dcoma,dataarray,gausscoeffs,zcoma):
+
+    #compute mean RA and DEC of current galaxy set
+    def part8a(dataarray):
+        #compute mean Right Ascension
+        sumnum = 0
+        sumden = 0
+        for i in range(0,len(dataarray)):
+            w = 10**(dataarray[i][4]/2.5)
+            sumnum += dataarray[i][1]/w
+            sumden += 1/w
+        meanra = sumnum/sumden
+
+        #compute mean Declination
+        sumnum = 0
+        sumden = 0
+        for i in range(0,len(dataarray)):
+            w = 10**(dataarray[i][4]/2.5)
+            sumnum += dataarray[i][2]/w
+            sumden += 1/w
+        meandec = sumnum/sumden
+        
+        print('Found mean RA,Dec: ',meanra,meandec)
+        return meanra,meandec
+
+    #perform gaussian fitting of current galaxy set
+    def part8b(dataarray):
+        
+        #create list of all redshifts
+        redshift = []
+        for i in range(0,len(dataarray)):
+            redshift.append(dataarray[i][6])
+
+        #Gather data in bins of width 0.001. Retrive counts from each bin. 
+        histogram = py.hist(redshift,bins=np.arange(0.005, 0.05, 0.001))
+        counts = histogram[0]
+        shifts = histogram[1]
+        binset = []
+
+        #Create list of z values at bin centers
+        for i in range(0,len(shifts)-1):
+            binset.append(shifts[i]+0.0005)
+
+        #Get initial parameter guesses (mean,stddev). Get gaussian fit.
```

Comment:
This is the second time (of four) that you define this exact same gaussian fit function. Define it up top (only once!) and just re-use it.